files/default/neutron-ha-tool.py

```diff
@@ -1061,48 +1061,53 @@ def pick(self):
         )
 
 
-class RemoteRouterNsCleanup(object):
-
-    def __init__(self, host, routerid):
+class RemoteNodeCleanup(object):
+    def __init__(self, host, timeout=10):
         self.target_host = host
-        self.namespace = "qrouter-" + routerid
-        self.timeout = 10
-        self.netns_del = "ip netns delete " + self.namespace
-        self.netns_pids = "ip netns pids " + self.namespace
+        self.timeout = timeout
+        self.netns_del = "ip netns delete "
+        self.netns_pids = "ip netns pids "
         self.netns_list = "ip netns list"
 
+    def _ssh_connect(self):
+        self.ssh_client = paramiko.SSHClient()
+        self.ssh_client.set_missing_host_key_policy(
+            paramiko.AutoAddPolicy())
+        self.ssh_client.load_system_host_keys()
+        self.ssh_client.connect(self.target_host, timeout=self.timeout)
+
     def _simple_ssh_command(self, command):
         # Note, that when get_pty is True, paramiko will never return anything
         # on the stderr channel, that's why we ignore it here. (stderr output
         # will endup in the stdout channel)
         _, stdout, _ = self.ssh_client.exec_command(command,
-                                                    timeout=10,
+                                                    timeout=self.timeout,
                                                     get_pty=True)
         out_lines = stdout.readlines()
         rc = stdout.channel.recv_exit_status()
         return [rc, [line.strip() for line in out_lines]]
 
-    def _namespace_exists(self):
+    def _namespace_exists(self, namespace):
```

Comment:
For example, look at https://github.com/SUSE-Cloud/cookbook-openstack-network/pull/27/files#diff-937ed0b9cba9efd2521c8fd1c9e8416fR402. @matelakat didn't *have* implement the `AgentList` class - he could have just used a `list` and put `calculate_maximal_router_count_difference` and related helper methods into one of the existing classes. But by choosing to separate these concerns into a new class rather than bloat an existing one, he ended up writing code which was beautifully clean, easy to read and maintain (and even unit-test, if that was necessary), for very little extra effort. Maybe I am missing a reason why this couldn't be done with a `Namespace` class?